tensorbay/opendataset/BDD100K/loader.py

```diff
@@ -124,34 +167,72 @@ def _BDD100K_loader(path: str, dataset_type: str) -> Dataset:
 
 
 def _load_segment(dataset: Dataset, root_path: str, dataset_type: str) -> None:
-    images_directory = os.path.join(root_path, "images", dataset_type)
     labels_directory = os.path.join(root_path, "labels")
 
-    get_data = _get_data_100k if dataset_type == "100k" else _get_data_10k
     read_label_file = _read_label_file_100k if dataset_type == "100k" else _read_label_file_10k
 
     for segment_name in _SEGMENT_NAMES:
         segment = dataset.create_segment(segment_name)
-        image_paths = _utility.glob(os.path.join(images_directory, segment_name, "*.jpg"))
+        image_paths = glob(os.path.join(root_path, "images", dataset_type, segment_name, "*.jpg"))
 
         print(f"Reading data to segment '{segment_name}'...")
         if segment_name == "test":
             for image_path in image_paths:
                 segment.append(Data(image_path))
         else:
+            if dataset_type == "10k":
+                single_channel_mask_paths: Dict[str, str] = {}
+                original_mask_paths: Dict[str, str] = {}
+                for seg_type, directory_names in _SEGMENTATIONS_INFO.items():
+                    original_mask_paths[seg_type] = os.path.join(
+                        labels_directory, *directory_names, segment_name
+                    )
+                    if seg_type != "sem":
+                        single_channel_mask_path = os.path.join(
+                            labels_directory,
+                            "single_channel_mask",
+                            segment_name,
+                            directory_names[0],
+                        )
+                        single_channel_mask_paths[seg_type] = single_channel_mask_path
+                        os.makedirs(single_channel_mask_path, exist_ok=True)
+
             label_contents = read_label_file(labels_directory, segment_name)
             for image_path in image_paths:
-                segment.append(get_data(image_path, label_contents[os.path.basename(image_path)]))
+                segment.append(
+                    _get_data_100k(image_path, label_contents[os.path.basename(image_path)])
+                    if dataset_type == "100k"
```

Comment:
there are too may checks about 100k and 10k in this function, can we split it into two functions?